clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -22882,6 +22882,7 @@ llvm::CallInst *CodeGenFunction::EmitFPBuiltinIndirectCall(
             .Case("sincos", llvm::Intrinsic::fpbuiltin_sincos)
             .Case("exp10", llvm::Intrinsic::fpbuiltin_exp10)
             .Case("rsqrt", llvm::Intrinsic::fpbuiltin_rsqrt)
+            .Case("__spirv_ocl_sin", llvm::Intrinsic::fpbuiltin_spirv_ocl_sin)
```

Comment:
I think this should just translate to the regular fpbuiltin intrinsic. The llvm-spirv translator will handle translating that back to the __spirv_ocl_ functions when needed.